src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs

```diff
@@ -475,8 +475,19 @@ private void Run()
                 (PowerShell pwsh, RunspaceInfo localRunspaceInfo, EngineIntrinsics engineIntrinsics) = CreateInitialPowerShellSession();
                 _mainRunspaceEngineIntrinsics = engineIntrinsics;
                 _localComputerName = localRunspaceInfo.SessionDetails.ComputerName;
-                _runspaceStack.Push(new RunspaceFrame(pwsh.Runspace, localRunspaceInfo));
-                PushPowerShellAndRunLoop(pwsh, PowerShellFrameType.Normal | PowerShellFrameType.Repl, localRunspaceInfo);
+
+                // NOTE: In order to support running events registered to PowerShell's OnIdle
+                // handler, we have to have our top-level PowerShell instance be nested (otherwise
+                // we get a PSInvalidOperationException because pipelines cannot be run
+                // concurrently). Specifically this bug cropped up when a profile loaded code which
+                // registered (and subsequently ran) on the OnIdle handler since it was hitting the
+                // non-nested PowerShell instance. So now we just start with a nested instance.
+                PowerShell nestedPwsh = CreateNestedPowerShell(localRunspaceInfo);
+                _runspaceStack.Push(new RunspaceFrame(nestedPwsh.Runspace, localRunspaceInfo));
+
+                // TODO: Should this now be marked as a Nested frame, even though for our purposes
+                // it isn't really nested?
```

Comment:
nah. Two very different meanings of nested. For us we mean nested executions, for PowerShell it basically equates to "on the pipeline thread"